pinot-spi/src/main/java/org/apache/pinot/spi/filesystem/FileInfo.java

```diff
@@ -0,0 +1,95 @@
+/**
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+package org.apache.pinot.spi.filesystem;
+
+import com.google.common.base.Preconditions;
+import java.util.Optional;
+import org.apache.commons.lang3.StringUtils;
+
+
+/**
+ * FileInfo contains the file path and many optional file attributes like mtime, length etc.
+ */
+public class FileInfo {
+  private final String _filePath;
+  private final Long _lastModifiedTime;
```

Comment:
Suggest making `_lastModifiedTime`, `_length` and `_isDirectory` primitive type to be consistent with other APIs. Essentially it should return the same info as calling each individual API

Reply:
I thought some future impl might not have all those attributes ready during file listing, so I've made them Optional.

Reply:
on a second thought, since there is the `String[] listFiles()` method, if some future impl can't fill those attributes during listing, then they should just just throw `Unsupported...Exception` and let callers use the basic listFiles()